### PR TITLE
Update blob_db_test

### PR DESCRIPTION
**Pull request by yiwu-arbug**
Summary:
I'm trying to improve unit test of blob db. I'm rewriting blob db test. In this patch:
* Rewrite tests of basic put/write/delete operations.
* Add disable_background_tasks to BlobDBOptionsImpl to allow me not running any background job for basic unit tests.
* Move DestroyBlobDB out from BlobDBImpl to be a standalone function.
* Remove all garbage collection related tests. Will rewrite them in following patch. 
* Disabled compression test since it is failing. Will fix in a followup patch.

Test Plan:
run the test.

### CODE REVIEW

File: utilities/blob_db/blob_db_impl.cc [hunk 1]
```diff
@@ -1060,6 +1036,7 @@ Status BlobDBImpl::PutUntil(const WriteOptions& options,
     value = CompressBlock(value_unc, compression_opts, &ct,
                           kBlockBasedTableVersionFormat, Slice(),
                           &compression_output);
+    printf("compression %lu -> %lu\n", value_unc.size(), value.size());
```
Comment (sagar0) [inline utilities/blob_db/blob_db_impl.cc]: left over from debugging?

File: utilities/blob_db/blob_db_impl.cc [hunk 2]
```diff
@@ -2244,6 +2221,39 @@ Iterator* BlobDBImpl::NewIterator(const ReadOptions& opts,
...
+  std::vector<std::string> filenames;
+  env->GetChildren(blobdir, &filenames);
+
+  for (const auto& f : filenames) {
+    uint64_t number;
+    FileType type;
+    if (ParseFileName(f, &number, &type) && type == kBlobFile) {
+      Status del = env->DeleteFile(blobdir + "/" + f);
+      if (status.ok() && !del.ok()) {
+        status = del;
```
Comment (sagar0) [inline utilities/blob_db/blob_db_impl.cc]: You would be discarding all subsequent non-ok statuses here, isn't it? Do you need to somehow handle them as well? 
Example: Let say `filenames` vector is of size 5. Each  `DeleteFile` failed with a different status/different reason. You enter line 2243 only for setting the first failure status, and never re-enter it again as status.ok() is false from here on. So, what is the plan for handling the subsequent non-ok statuses?

File: utilities/blob_db/blob_db_impl.cc [hunk 3]
```diff
@@ -2244,6 +2221,39 @@ Iterator* BlobDBImpl::NewIterator(const ReadOptions& opts,
...
+      }
+    }
+  }
+  env->DeleteDir(blobdir);
+
+  Status destroy = DestroyDB(dbname, options);
+  if (status.ok() && !destroy.ok()) {
+    status = destroy;
```
Comment (sagar0) [inline utilities/blob_db/blob_db_impl.cc]: Same comment as above. Execution never reaches 2251 if there was `DeleteFile` already failed in line 2241. `destroy` status is just discarded. 
How important is it to return multiple failure statuses from this `DestroyBlobDB` function to the caller?

Comment (sagar0) [inline utilities/blob_db/blob_db_impl.cc]: Btw, I am ok with addressing this issue separately, may be in a different PR, as the current PR is not related to this issue.

File: utilities/blob_db/blob_db_impl.cc (hunk 2 above)
Comment (yiwu-arbug) [inline utilities/blob_db/blob_db_impl.cc]: I'm trying to match what `DestroyDB()` is doing: https://github.com/facebook/rocksdb/blob/master/db/db_impl.cc#L2229-L2231
I think the idea is to have `DestroyDB()` try to delete every file of the database regardless of errors.

File: utilities/blob_db/blob_db_test.cc [hunk 4]
```diff
@@ -19,65 +19,51 @@
...
-}
 
 class BlobDBTest : public testing::Test {
  public:
   const int kMaxBlobSize = 1 << 14;
 
-  BlobDBTest() : blobdb_(nullptr) {
-    dbname_ = test::TmpDir() + "/blob_db_test";
-    // Reopen(BlobDBOptionsImpl());
+  BlobDBTest() : dbname_(test::TmpDir() + "/blob_db_test"), blob_db_(nullptr) {
```
Comment (anirbanr-fb) [inline utilities/blob_db/blob_db_test.cc]: unnecessary change?
